vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -602,9 +609,39 @@ - (void)reportNewTabPageAdEvent:(NSString*)wallpaperId
                            static_cast<ads::NewTabPageAdEventType>(eventType));
 }
 
+- (void)inlineContentAdsWithDimensions:(NSString*)dimensions,
+                            completion:
+                                (void (^)(BOOL success,
+                                          BATInlineContentAd* ad))completion {
+  if (![self isAdsServiceRunning]) {
+    return;
+  }
+  ads->GetInlineContentAd(
+      dimensions.UTF8String,
+      ^(const bool success, const ads::InlineContentAdInfo& ad) {
+        const auto inline_content_ad =
+            [[BATInlineContentAd alloc] initWithInlineContentAdInfo:ad];
```

Comment:
What happens to `ad` here if `success` is false? Is it passing an empty object? Or is it `nullptr`?

Reply:
empty object